### PR TITLE
Editor Interface: Remove  small header from global styles/plugin sidebar

### LINKED ISSUE #64095: Address duplication caused by ComplementaryAreaHeader
On small screens, panels in the Editor show a heading via `ComplementaryAreaHeader`. This produces some unnecessary duplication:

Screenshot 2024-07-30 at 13 15 39

In both cases the heading feels superfluous. The close button is debatable, but given the buttons in the top bar act as toggles, they also feel unnecessary. 

Should these panels be updated to omit `ComplementaryAreaHeader`?

### PR DESCRIPTION
**Pull request by t-hamano**
Fixes #64095, #62067

## What?

~~This PR removes the small screen header from the `ComplementaryAreaHeader` component. This means that the `smallScreenTitle` prop on the `ComplementaryArea` component will be removed.~~

This PR removes the small screen header from the global styles sidebar and plugin sidebar.

## Why?

As mentioned in #62067, it seems that a header for small screens was necessary up until WP6.3:

334421685-b324e49b-730c-47b2-834e-0a09157758c2

However, now that the default header has exactly the same functionality, it seems that a header for small screens is no longer necessary.

## How?

This PR also exposes some header buttons (revisions, more menu) that were intentionally hidden on smaller screens.

This makes the following features available on smaller screens:

- Global Syles Revisions
- Additional CSS
- Reset styles

I initially thought about removing the `smallScreenTitle` prop from the `ComplementaryArea` component, but it seems that there are plugins that use the `ComplementaryArea` component directly to build their own sidebars.

- [Search results in WP Directory](https://wpdirectory.net/search/01J68V0Q3GR84XTP1RTVKF52WR)
- [`smallScreenTitle` prop defined in the MailPoet plugin](https://github.com/search?q=org%3Amailpoet%20smallScreenTitle&type=code)

~~I decided to keep this prop because these plugins may still want to provide the title for mobile:~~

image

**Update**: This component is not exposed in the global wp object. Developers must explicitly bundle the interface package. This means that unless developers update their interface package, it will not be affected by this PR. Therefore, we decided to remove the `smallScreenTitle` prop.

## Testing Instructions

Verify that the following screens are displayed correctly on a small screen and that the newly exposed features work.

- Block inspector
- Global Styles
  - Global Styles Revisions
  - Additional CSS
  - Reset CSS
- PluginsSidebar

### Block Inspector

| Before | After |
|--------|--------|
| image | image | 

### Global Styles

| Before | After |
|--------|--------|
| image| image | 

### PluginsSidebar

Run the following code in your browser console:

<details><summary>Details</summary>

```javascript
wp.plugins.registerPlugin( 'my-plugin-sidebar', {
	render: function () {
		return React.createElement(
			wp.editor.PluginSidebar,
			{
				name: 'my-plugin-sidebar',
				icon: 'admin-post',
				title: 'My plugin sidebar',
			},
			'Content'
		);
	},
} );
```

</details> 

| Before | After |
|--------|--------|
| image | image| 

### Global Styles Revisions

| Before | After |
|--------|--------|
| - | image| 

### Additional CSS

| Before | After |
|--------|--------|
| - | image|

### CODE REVIEW

File: packages/interface/src/components/complementary-area/README.md
Comment (t-hamano) [inline packages/interface/src/components/complementary-area/README.md:93]: I'm wondering if we should simply remove this prop, or show a deprecation message.

**Comment by jameskoster** [comment]
Thank you, this is all working as advertised for me.

**Comment by jasmussen** [comment]
Thanks so much for working on this, this is an important one! 🙏

File: packages/interface/src/components/complementary-area/README.md
Comment (stokesman) [reply packages/interface/src/components/complementary-area/README.md:93]: I like simply removing it but wouldn’t say I know best.

**Review by stokesman — APPROVED** [review]
This makes for a nice cleanup in UI and code and I spotted nothing out of place in either. Nicely done.

I'm sure you may wish to wait for a second opinion—at the least about the decision to remove `smallScreenTitle` without deprecation.

**Comment by t-hamano** [comment]
@stokesman Thanks for the feedback!

After thinking about it, I realized that developers who use the `ComplementaryArea` component directly might want to display the title for mobile as before. Therefore, I tried an approach that maintains this prop. Details are described in the updated description.

What do you think?

**Comment by stokesman** [comment]
> I realized that developers who use the `ComplementaryArea` component directly might want to display the title for mobile as before.

Perhaps. The caution is good and may be the side to err on. Still, I don’t think I can see utility in the `smallScreenTitle` prop. It seems like the `header` prop is flexible enough that it could fulfill any requirement to display differently on mobile.

Another consideration that may factor in is that a plugin consuming the interface package has to bundle it and so cannot be directly broken by new releases of the package. Maybe that permits moving forward with the removed prop. If so, a dev note seems a good idea along with a changelog entry.

---- 

Totally separate but I wonder what’s up with the excessive line-height as seen in your screenshot (at "Add your own CSS…of your site…"):

image

I’m curious as it looks okay to me on Chrome/MacOS.

**Comment by t-hamano** [comment]
> a plugin consuming the interface package has to bundle it and so cannot be directly broken by new releases of the package.

I didn't know this! The `@wordpress/interface` package, like many other packages, is automatically bundled into the global `wp` object, so I thought that a WordPress update would automatically update the `@wordpress/interface` package as well.

If that's the case, adding a changelog and writing a dev note would be enough.

> Totally separate but I wonder what’s up with the excessive line-height as seen in your screenshot (at "Add your own CSS…of your site…"):

The problem turned out to be caused by the link having a top margin. The issue wasn't apparent on the desktop view, but the PR rexposes additional CSS for mobile, so it's causing an issue:

link-margin-top

I solved it by adding the `br` element before the link:

image

**Comment by stokesman** [comment]
> I didn't know this!

I’d forgotten it until I tested the MailPoet plugin on this branch then noticed the changes don’t affect it and had to remember why.

> The `@wordpress/interface` package, like many other packages, is automatically bundled into the global `wp` object, so I thought that a WordPress update would automatically update the `@wordpress/interface` package as well.

It‘s true that a WordPress update will have the latest `@wordpress/interface` bundled in just that it isn’t exposed in the global `wp` object and therefore a plugin can’t be using it.

Nice that you fixed the little issue making the line-height look off. Makes sense.

This LGTM again. You may wish to have another member give it a sanity check though.

**Comment by afercia** [comment]
Tests well for me. 

Screenshot before:

before

After:

after

**Comment by Mamaduka** [comment]
Closed #62067 manually.

GitHub only recognizes the first issue after the unique keyword. If you want to link multiple issues to a single PR, use the keyword for each one.

Example:

```
Fixes #A.
Fixes #B.
```

**Pull request merged**
